.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,86 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '22 12 * * 2'
```

Comment:
Could you elaborate more on why we are only requiring this github action on main branch for pushes and pull-requests? 

Also why a cron job is needed/

Reply:
The running time for this is too long, I'd rather have this be run manually

Reply:
Could you collaborate more on the effect of CodeQL?

Reply:
Don't really know, that is why it says trying out codeql. I found it in the security tab of git
![image](https://user-images.githubusercontent.com/42769475/182520887-99bef700-8137-4029-b03b-bd600f8fc156.png)
I think it essentially scans your code for all kinds of security vulnerabilities.